Backpack/Skeleton/Classes/BPKSkeleton.swift

```diff
@@ -0,0 +1,162 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+import UIKit
+import SwiftUI
+
+public class BPKSkeleton: UIView {
+    static internal var bgColor: UIColor =
+        BPKColor.dynamicColor(withLightVariant: BPKColor.skyGrayTint06, darkVariant: BPKColor.blackTint02)
+    static internal var defaultColor: UIColor =
+        BPKColor.dynamicColor(withLightVariant: BPKColor.white, darkVariant: BPKColor.black)
+    private var customSize: CGSize = .zero
+    
+    public var type: BPKSkeletonType = .image {
+        didSet {
+            updateType()
+        }
+    }
+    
+    public var size: BPKSkeletonSize = .default
+    
+    public var style: BPKSkeletonStyle = .default {
+        didSet {
+            updateStyle()
+        }
+    }
+    
+    private var skeletonView: UIView = {
+        let view = UIView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.clipsToBounds = true
+        return view
+    }()
+    
+    public convenience init(
+        type: BPKSkeletonType,
+        size: BPKSkeletonSize = .default,
+        style: BPKSkeletonStyle = .default
+    ) {
+        self.init()
+        self.type = type
+        self.size = size
+        self.style = style
+        
+        updateType()
+        setup()
+    }
+    
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+    }
+    
+    public required init?(coder: NSCoder) {
+        super.init(coder: coder)
+    }
```

Comment:
Should these call setup as well? If we don't call setup, then we might miss setting up the skeleton.

Reply:
I think yes.
Also not sure if a convenience init is a good choice here. having the designited initializer with default values may be cleaner and remove the need to have 3 inits defined

```swift
public init(
  type: BPKSkeletonType = .image,
  size: BPKSkeletonSize = .default,
  style: BPKSkeletonStyle = .default
) {
  self.type = type
  self.size = size
  self.style = style
  updateType() // <--- Should this method be called inside setup()? probably yes
  setup()
  super.init(frame: frame)
}

public required init?(coder: NSCoder) {
  super.init(coder: coder)
  updateType()
  setup()
}